sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTest.cs

```diff
@@ -118,25 +116,49 @@ public void ThreadCGUD_MemberAUR_MessageGSU_NotificationT()
             var getMessagesCountThread2 = messagesThread2.Count();
 
             # region Pagination assertions
-            Pageable<ChatMessage> messagesPaginationTest = chatThreadClient.GetMessages();
+            Pageable<ChatMessage> allMessages = chatThreadClient.GetMessages();
             string? continuationToken = null;
             var expectedPageSize = 2;
             var messagesTotal = 0;
-            var messagesCounter = 0;
+            var actualMessagesPerPage = 0;
+            var sentTextMessageIndex = messagesMapping.Length;
+            DateTimeOffset? previousCreatedOn = DateTimeOffset.MaxValue;
             ulong previousId = ulong.MaxValue;
-            foreach (Page<ChatMessage> messagesPage in messagesPaginationTest.AsPages(continuationToken, expectedPageSize))
+            foreach (Page<ChatMessage> messagesPage in allMessages.AsPages(continuationToken, expectedPageSize))
             {
-                (messagesCounter, continuationToken) = AssertPageableChatMessages(messagesPage, expectedPageSize, messagesMapping);
-                messagesTotal+= messagesCounter;
-
-                //Make sure expected order is ok
+                actualMessagesPerPage = messagesPage.Values.Count;
                 foreach (ChatMessage chatMessage in messagesPage.Values)
                 {
+                    if (chatMessage.Type == "Text")
+                    {
+                        sentTextMessageIndex--;
+                        if (sentTextMessageIndex == 1)
+                            sentTextMessageIndex--; // skip index for message on thread 2
+                        Assert.AreEqual(messagesMapping[sentTextMessageIndex].Content, chatMessage.Content);
+                        Assert.AreEqual(messagesMapping[sentTextMessageIndex].Id, chatMessage.Id);
+                    }
+                    // Make sure all message types (not just TEXT) are kept in the expected order received from the service. Equals needs to be accounted due to ms precision conversion.
+                    Assert.LessOrEqual(chatMessage.CreatedOn, previousCreatedOn);
+
+                    // Message Ids are auto-incremented / uniquely created downstream using epoc time conversions. Could be used on the client side to disambiguate order in case of identical CreatedOn timestamps
```

Comment:
I do not think we should have the logic about sorting by id, testing that should be done on the service level. Developers should assume that the response is already sorted and not have to worry about ids.